src/main/java/fr/dynamx/client/renders/model/renderer/ObjItemModelLoader.java

```diff
@@ -62,12 +62,12 @@ public void renderByItem(ItemStack stack, float partialTicks) {
                 return;
                 //throw new NoSuchElementException("Item " + stack.getItem().getRegistryName() + " is not registered with metadata " + stack.getMetadata() + " into ObjItemModelLoader");
             }
-            if (!MinecraftForge.EVENT_BUS.post(new DynamXItemEvent.Render(stack.getItem(), EventStage.PRE, ItemCameraTransforms.TransformType.NONE))) {
+            if (!MinecraftForge.EVENT_BUS.post(new DynamXItemEvent.Render(stack, EventStage.PRE, renderType))) {
                 if (model.getOwner().get3DItemRenderLocation() == Enum3DRenderLocation.NONE || (renderType == ItemCameraTransforms.TransformType.GUI && model.getOwner().get3DItemRenderLocation() == Enum3DRenderLocation.WORLD)) {
                     GlStateManager.translate(0.5F, 0.5F, 0.5F);
                     Minecraft.getMinecraft().getRenderItem().renderItem(stack, model.getGuiBaked());
                 } else {
-                    if (!MinecraftForge.EVENT_BUS.post(new DynamXItemEvent.Render(stack.getItem(), EventStage.PRE, renderType)))
+                    if (!MinecraftForge.EVENT_BUS.post(new DynamXItemEvent.Render(stack, EventStage.PRE, renderType)))
```

Comment:
I think it would be good to introduce a EventStage.TRANSFORM, and encapsulate all transforms in this event (lines 71 to 79)

(for now, it's the exact same event as at line 65)